examples/api/lib/material/bottom_app_bar/bottom_app_bar.2.dart

```diff
@@ -0,0 +1,192 @@
+  runApp(const BottomAppBarDemo());
+}
+
+class BottomAppBarDemo extends StatefulWidget {
+  const BottomAppBarDemo({super.key});
+
+  @override
+  State createState() => _BottomAppBarDemoState();
+}
+
+class _BottomAppBarDemoState extends State<BottomAppBarDemo> {
+  late ScrollController _controller;
+
+  bool _showFab = true;
+  bool _isElevated = true;
+  bool _isVisible = true;
+
+  FloatingActionButtonLocation get _fabLocation => _isVisible
+      ? FloatingActionButtonLocation.endContained
+      : FloatingActionButtonLocation.endFloat;
+
+  void _onShowFabChanged(bool value) {
+    setState(() {
+      _showFab = value;
+    });
+  }
+
+  void _onElevatedChanged(bool value) {
+    setState(() {
+      _isElevated = value;
+    });
+  }
+
+  static const List<Color> colors = <Color>[
+    Colors.yellow,
+    Colors.orange,
+    Colors.pink,
+    Colors.purple,
+    Colors.cyan,
+  ];
+
+  static final List<Widget> items = List<Widget>.generate(
+    colors.length,
+    (int index) => Container(color: colors[index], height: 150.0),
+  ).reversed.toList();
+
+  void _listen() {
+    final ScrollDirection direction = _controller.position.userScrollDirection;
+    if (direction == ScrollDirection.forward) {
+      _show();
+    } else if (direction == ScrollDirection.reverse) {
+      _hide();
+    }
+  }
+
+  void _show() {
+    if (!_isVisible) {
+      setState(() => _isVisible = true);
+    }
+  }
+
+  void _hide() {
+    if (_isVisible) {
+      setState(() => _isVisible = false);
+    }
+  }
+
+  void _addNewItem() {
+    setState(() {
+      items.insert(
+        0,
+        Container(
+          height: 150.0,
+          color: colors[items.length % 5],
+        ),
+      );
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = ScrollController();
+    _controller.addListener(_listen);
+  }
+
+  @override
+  void dispose() {
+    _controller.removeListener(_listen);
+    _controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      theme: ThemeData(useMaterial3: true),
+      home: Scaffold(
+        appBar: AppBar(
+          title: const Text('Bottom App Bar Demo'),
+        ),
+        body: ListView(
+          controller: _controller,
+          children: <Widget>[
+            SwitchListTile(
+              title: const Text('Floating Action Button'),
+              value: _showFab,
+              onChanged: _onShowFabChanged,
+            ),
+            SwitchListTile(
+              title: const Text('Bottom App Bar Elevation'),
+              value: _isElevated,
+              onChanged: _onElevatedChanged,
+            ),
+            ...items,
+          ],
+        ),
+        floatingActionButton: _showFab
+            ? FloatingActionButton(
+                onPressed: () => _addNewItem(),
+                tooltip: 'Add new item',
+                child: const Icon(Icons.add),
+              )
```

Comment:
```suggestion
            ? FloatingActionButton(
                onPressed: () => _addNewItem(),
                elevation: _isVisible ? 0 : null,
                tooltip: 'Add new item',
                child: const Icon(Icons.add),
              )
```

Which would handle the elevation change.